### PR TITLE
New conformance and e2e tests

### PR DESCRIPTION
**Pull request by aavarghese**
Signed-off-by: Ansu Varghese <avarghese@us.ibm.com>

Fixes #

## Proposed Changes

- Conformance test support for Redis Stream Source. To run them:
```
ko apply -Rf test/config/
export SYSTEM_NAMESPACE=knative-eventing

go test -v -tags=e2e,source -count=1 ./test/conformance -sources=sources.knative.dev/v1alpha1:RedisStreamSource
or
./test/e2e-conformance-tests.sh
```

- E2E test support for Redis Stream Source and Redis Stream Sink. To run them:
```
export TEST_REDISSTREAMSOURCE=1 
export TEST_REDISSTREAMSINK=1
test/e2e-tests.sh --run-tests
```

- E2E/Integration test support for Redis Sink together with Redis Source, consuming the event added to Redis by the Redis Sink. To run them (after unsetting the env vars from above):
```
export TEST_REDISSTREAMINTEGRATION=1
test/e2e-tests.sh --run-tests
```

- Performance tests (coming soon)

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: sink/config/300-redisstreamsink.yaml [hunk 1]
```diff
@@ -21,6 +21,7 @@ metadata:
     eventing.knative.dev/release: devel
     knative.dev/crd-install: "true"
     duck.knative.dev/addressable: "true"
+    duck.knative.dev/source: "true"
```
Comment (lionelvillard) [inline sink/config/300-redisstreamsink.yaml:24]: A sink does not implement the source duck type.

Comment (aavarghese) [reply sink/config/300-redisstreamsink.yaml:24]: Oh oops, you're right. This shouldn't be added. Thanks!

Comment (aavarghese) [reply sink/config/300-redisstreamsink.yaml]: Done

**Comment by aavarghese** [comment]
@lionelvillard fixed sink as per your comment. And added a new e2e test for sink+source working together. Could you review please?

File: sink/README.md [hunk 2]
```diff
@@ -22,14 +22,14 @@ receiver, if they don't already exist.
   using a cloud instance of Redis (for example, Redis DB on IBM Cloud), a TLS
   certificate will need to be configured, prior to installing the event sink.
 
-      Edit the [`config-tls`](config/config-tls.yaml) Config Map to add the TLS Certicate
-      from your cloud instance of Redis to the `cert.pem` data key:
+  Edit the [`config-tls`](config/config-tls.yaml) Config Map to add the TLS Certicate
```
Comment (lionelvillard) [inline sink/README.md:25]: nit: Config Map -> ConfigMap

File: source/config/201-clusterrole.yaml [hunk 3]
```diff
@@ -97,7 +97,7 @@ rules:
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: eventing-redis-source-observer
+  name: eventing-sources-source-observer
```
Comment (lionelvillard) [inline source/config/201-clusterrole.yaml:100]: this should be `eventing-redis-source-observer`

File: sink/README.md (hunk 2 above)
Comment (aavarghese) [reply sink/README.md:25]: Done

File: source/config/201-clusterrole.yaml (hunk 3 above)
Comment (aavarghese) [reply source/config/201-clusterrole.yaml:100]: Today, we have a [source conformance test](eventing-sources-source-observer) that checks for this particular name and if not, fails the test. For a source, it expects only a clusterrole with name `eventing-sources-source-observer` to be valid and its rules be added to `source-observer` role.

**Pull request closed without merging**